armory/scenarios/scenario.py

```diff
@@ -254,11 +253,15 @@ def evaluate_all(self):
         for _ in tqdm(range(len(self.test_dataset)), desc="Evaluation"):
             self.next()
             self.evaluate_current()
+        self.hub.set_context(stage="finished")
 
     def next(self):
+        self.hub.set_context(stage="next")
```

Comment:
for my knowledge/curiosity: what's the purpose behind setting stage to `"next"` as well as `"finished"`?

Reply:
Mainly so that we don't accidentally tag things outside of `run_attack` or `run_benign` as `benign` or `attack` or `adversarial`.